app/controllers/users.js

```diff
@@ -15,8 +23,16 @@ exports.createUser = async (req, res) => {
 
 exports.createUserAdmin = async (req, res) => {
   try {
-    const dataUser = await saveUserAdmin(signUp(req.body));
-    res.status(200).send(dataBasic(dataUser));
+    let response = {};
+    const dataNewUser = signUp(req.body);
+    const user = await getUserByEmail(dataNewUser.email);
+    if (user) {
+      response = await updateUserByEmail(user.email, { role: roles.ADMIN });
+    } else {
```

Comment:
Instead of the else, I think it would be better to return directly in the if in case the user is updated and in case the user is not there, ignore the if and create the user and return, also remember to log information about what happens